nextcloud_install_production.sh

```diff
@@ -2,6 +2,71 @@
 
 # T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
 
+# Variables:
+# Todo: make DOWNLOAD_DIR readable again; is disabled for now to fix travis
+SCRIPTS=/var/scripts
+#DOWNLOAD_DIR=/tmp/NC_VM
+
+# Download and validate Nextcloud VM
+if ! [ -d $SCRIPTS ]
+then
+    # Check if root
+    if [[ "$EUID" -ne 0 ]]
+    then
+        echo "You have to run this script as root! Exiting..."
+        exit 1
+    fi
+
+    # Test internet connection
+    if ! ping -c1 -W1 github.com &>/dev/null
+    then
+        echo "Couldn't reach github.com. Exiting..."
+        exit 1
+    fi
+
+    # Download the release package (.tar.gz) file
+    # Todo: Remove the hashtags after it is released and change the download path if necessary
+    #rm -rf $DOWNLOAD_DIR
+    #mkdir $DOWNLOAD_DIR
+    #cd $DOWNLOAD_DIR
+    #curl -sLO https://github.com$(curl -s -m 900 https://github.com/nextcloud/vm/tags/ | grep -om1 "/nextcloud/vm/archive/[0-9.]*.tar.gz")
+
+    # Todo: Verify the state of the downloaded package with a checksum
+
+    # Todo: Verify the release package and integrity with the gpg key?
+
+    # Extract everything to "$SCRIPTS"
+    # Todo: Remove the hashtags after it is released
+    #VM_VERSION=$(ls $DOWNLOAD_DIR)
+    #tar -xvzf $DOWNLOAD_DIR/$VM_VERSION -C /var
+    #mv /var/vm-${VM_VERSION%%.tar.gz} $SCRIPTS
+    #rm -rf $DOWNLOAD_DIR
+
+    # This section is for testing purposes only; should get removed if everything above is ready.
+    git clone -b run_locally --single-branch https://github.com/nextcloud/vm.git "$SCRIPTS" &>/dev/null
+
+    # Remove all unnecessary files
+    rm -rf "$SCRIPTS"/.git
+    rm "$SCRIPTS"/LICENSE
+    rm "$SCRIPTS"/issue_template.md
+    rm "$SCRIPTS"/.travis.yml
+    rm "$SCRIPTS"/README.md
+
+    # Move all main files to "$SCRIPTS"/main (apart from install-production and startup-script)
+    mkdir -p "$SCRIPTS"/main
+    mv "$SCRIPTS"/lib.sh "$SCRIPTS"/main
+    mv "$SCRIPTS"/nextcloud_update.sh "$SCRIPTS"/main
+
+    # Set ownership and permission
+    chown -R root:root "$SCRIPTS"
+    chmod -R +x "$SCRIPTS"
+
+    # Run the nextcloud_install_production script in a new process
+    exec "$SCRIPTS"/nextcloud_install_production.sh
+fi
+
+# -----------------------------------------------------------------------------------------------------------------------------------
```

Comment:
I know in this part are still some Todos left but they will only work after a new release. For now it gets the scripts directly from the branch. Later on they should be downloaded from the archive: see the "#Download the release package" section. The order is the following: remove the hashtatgs in that section, make a new release and it will work.

The big advantage is that the initial installation will work no matter if the network works or not since everything is already downloaded during the startup script.

We can add a checksum and possibly sign the release package later on, it is planned but not necessary for this feature to work.